src/services/serviceProvider.ts

```diff
@@ -161,6 +163,43 @@ export class ServiceProvider {
     ServiceProvider.serviceMap.clear();
   }
 
+  /**
+   * Checks if a command is available in the list of VSCode commands.
+   * @param command - The command to check.
+   * @throws {Error} If the command is not available.
+   */
+  private static async checkCommandAvailability(
+    command: string
+  ): Promise<void> {
+    const availableCommands = await this.getCommands();
+    if (!availableCommands.includes(command)) {
+      throw new Error(
+        `Command ${command} cannot be found in the current vscode session.`
+      );
+    }
+  }
+
+  private static getCommands(): Promise<string[]> {
```

Comment:
I had to add a simple method that I could mock to cover the actual call to vscode.commands.getCommands. Could not get a jest mock to properly hand the getCommands function.